### PR TITLE
AGD-1983 : allow logs in cli mode

### PR DESCRIPTION
**Pull request by BogdanZavu**
### Purpose
We want to allow logging while in CLI mode as well. This is more important now because the CLI will be used by Generative Design.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### FYIs
@saintentropy

### CODE REVIEW

File: src/DynamoCore/Logging/DynamoLogger.cs
Comment (mjkkirschner) [inline src/DynamoCore/Logging/DynamoLogger.cs:212]: because you now are allowing arbitrary log directories, maybe inside `StartLoggingToConsoleAndFile` we should check that we have write access to the directory and that it exists?

File: src/DynamoCore/Logging/DynamoLogger.cs
Comment (mjkkirschner) [inline src/DynamoCore/Logging/DynamoLogger.cs:205]: maybe add some comments on what this controls?

File: src/DynamoCore/Logging/DynamoLogger.cs
Comment (BogdanZavu) [reply src/DynamoCore/Logging/DynamoLogger.cs:212]: So nothing really changes, the log dir is the same as before.

**Comment by BogdanZavu** [comment]
@mjkkirschner there is 1 failed test but doesn't look related to my change

**Review by mjkkirschner — APPROVED** [review]
@BogdanZavu I agree, LGTM - merge when you like.

**Comment by BogdanZavu** [comment]
@mjkkirschner it seems that the merge button is disabled ( for me at least ) when the selfserve has failed. Can you merge it ?

**Pull request merged**
